### PR TITLE
async/awaitify remaining logic in webusb.ts, tweak error message on locked device

### PR DESCRIPTION
**Pull request by jwunderl**
Was walking through this code line by line on friday to try and figure out what was up with chromebook webusb, so async/await-ified the remaining bits along the way. Also added some type info to navigator to remove some unnecessary casting, & made the 'connection failure' dialog a bit more explicit in cases where user gave a device but it didn't work (as that most often comes down to them having it already open in another browser / application, e.g. incognito + normal browser at once).

### CODE REVIEW

File: pxtlib/webusb.ts [hunk 1]
```diff
@@ -197,40 +225,38 @@ namespace pxt.usb {
...
         }
 
         disable() {
             if (!this.enabled) return;
 
             this.enabled = false;
             this.log(`unregistering webusb events`);
-            (navigator as any).usb.removeEventListener('disconnect', this.handleUSBDisconnected);
-            (navigator as any).usb.removeEventListener('connect', this.handleUSBConnected);
+            navigator?.usb.removeEventListener('disconnect', this.handleUSBDisconnected);
```
Comment (eanders-ms) [inline pxtlib/webusb.ts:237]: `usb` is declared optional in the interface. Do you want to honor that here?
```suggestion
            navigator?.usb?.removeEventListener('disconnect', this.handleUSBDisconnected);
```

File: pxtlib/webusb.ts [hunk 2]
```diff
@@ -421,104 +447,101 @@ namespace pxt.usb {
...
+                this.log("using ctrl pipe");
+            }
+            this.log("claim interface");
+            await dev.claimInterface(iface.interfaceNumber);
+            this.log("device ready");
+            this.lastKnownDeviceSerialNumber = this.dev.serialNumber;
+            this.ready = true;
+            if (isHF2) {
+                // just starting, not waiting on it.
+                this.readLoop();
```
Comment (eanders-ms) [inline pxtlib/webusb.ts:510]: For extra clarity on intent here:
```suggestion
                /*await*/ this.readLoop();
```

File: pxtlib/webusb.ts [hunk 3]
```diff
@@ -421,104 +447,101 @@ namespace pxt.usb {
...
-            .then(dev => !!dev)
-            .catch(e => {
-                // user cancelled
-                if (e.name == "NotFoundError")
-                    return undefined;
-                throw e;
-            })
+    export async function pairAsync(): Promise<boolean> {
+        try {
+            const dev = await navigator.usb.requestDevice({
```
Comment (eanders-ms) [inline pxtlib/webusb.ts:518]: Could safe continuation be needed here? There are a few other instances in the file as well.
```suggestion
            const dev = await navigator?.usb?.requestDevice({
```

**Review by eanders-ms — APPROVED** [review]
Nice cleanup. Made a few minor suggestions.

File: pxtlib/webusb.ts (hunk 1 above)
Comment (jwunderl) [reply pxtlib/webusb.ts:237]: Whoops, i intended to put them there yeah, usb feature will be disabled overall if `navigator.usb` is unset so this was for clarity but fixing, thanks!

File: pxtlib/webusb.ts [hunk 4]
```diff
@@ -197,40 +225,38 @@ namespace pxt.usb {
 
             this.enabled = true;
             this.log("registering webusb events");
-            (navigator as any).usb.addEventListener('disconnect', this.handleUSBDisconnected, false);
-            (navigator as any).usb.addEventListener('connect', this.handleUSBConnected, false);
+            navigator.usb?.addEventListener('disconnect', this.handleUSBDisconnected, false);
```
Comment (srietkerk) [inline pxtlib/webusb.ts:228]: Similar to Eric's questions about safe continuation. Does this need 
```suggestion
            navigator?.usb?.addEventListener('disconnect', this.handleUSBDisconnected, false);
```
like there are in other places or is just having `.usb` fine in this case? Same question applies to the line below.

File: pxtlib/webusb.ts (hunk 3 above)
Comment (jwunderl) [reply pxtlib/webusb.ts]: https://github.com/microsoft/pxt/blob/dev/jwunderl/async-ify-webusb-cleanup/pxtlib/webusb.ts#L608 is where it'll be checked overall, but adding in here for clarity sake ~

File: pxtlib/webusb.ts (hunk 4 above)
Comment (jwunderl) [reply pxtlib/webusb.ts:228]: https://github.com/microsoft/pxt/blob/dev/jwunderl/async-ify-webusb-cleanup/pxtlib/webusb.ts#L608 is where we check for it being enabled for the `pxt.usb.isEnabled()` which we use to gate all other calls; but adding in for clarity sake~

**Review by srietkerk — APPROVED** [review]

**Pull request merged**